src/schema.c

```diff
@@ -198,32 +353,31 @@ ldap_schema_append_objectclass(struct ldap_schema_t *schema, LDAPObjectClass *ob
             ld_error("Attempt to pass NULL object class parameter. \n");
         }
 
+        if (!objectclass_names)
+        {
+            ld_error("Object class names list is empty!\n");
+        }
+
         return false;
     }
 
-    if (schema->object_classes_size >= schema->object_classes_capacity - 1)
+    bool result = g_hash_table_insert(schema->attribute_types_by_oid, objectclass->oc_oid, objectclass);
+
+    if (!result)
     {
-        int required_capacity = schema->object_classes_capacity * 2;
-        TALLOC_CTX* ctx = talloc_parent(schema);
-        LDAPObjectClass** classes = talloc_realloc(ctx,
-                                                   schema->object_classes,
-                                                   LDAPObjectClass*,
-                                                   required_capacity);
-        if (!classes)
+        return false;
+    }
+
+    for (int i = 0; objectclass_names[i] != NULL; ++i)
+    {
+        result = result && g_hash_table_insert(schema->attribute_types_by_name, objectclass_names[i], objectclass);
```

Comment:
We do not need `result &&` here because we want to check result of the last operation here.